lib/git.js

```diff
@@ -4,7 +4,7 @@ var Bluebird = require('bluebird');
 var CP       = Bluebird.promisifyAll(require('child_process'));
 
 var SEPARATOR      = '===END===';
-var COMMIT_PATTERN = /^(\w*)(\(([\w\$\.\-\* ]*)\))?\: (.*)$/;
+var COMMIT_PATTERN = /^(\w*)(?:\(([^)]*?)\)|):\s*(.*?(?:\[([^\]]+?)\]|))$/;
```

Comment:
I can make this even a bit more flexible by allowing the flags to occur anywhere in the subject line.

Reply:
I think it might be better to enforce it only at the end. I'd rather it be well-defined on when it'll be picked up as opposed to unintentionally being picked up.

Reply:
👍